src/sts.rs

```diff
@@ -0,0 +1,278 @@
+//! The AWS STS API.
+//!
+//! If you're using the service, you're probably looking for [StsClient](struct.StsClient.html).
+#![allow(unused_imports)]
+
+include!(concat!(env!("OUT_DIR"), "/sts.rs"));
+
+pub use self::credential::{
+    StsSessionCredentialsProvider,
+    StsAssumeRoleSessionCredentialsProvider,
+    StsWebIdentityFederationSessionCredentialsProvider,
+    NewAwsCredsForStsCreds,
+    };
+
+mod credential {
+    use std::error::Error;
+    use std::fmt;
+
+    use hyper::Client;
+    use chrono::*;
+
+    use ::{AwsCredentials, ProfileProvider, CredentialsError,
+        ProvideAwsCredentials, DispatchSignedRequest};
+    use super::{AssumeRoleRequest, AssumeRoleResponse, AssumeRoleError,
+        AssumeRoleWithSAMLRequest, AssumeRoleWithSAMLResponse, AssumeRoleWithSAMLError,
+        AssumeRoleWithWebIdentityRequest, AssumeRoleWithWebIdentityResponse, AssumeRoleWithWebIdentityError,
+        DecodeAuthorizationMessageRequest, DecodeAuthorizationMessageResponse, DecodeAuthorizationMessageError,
+        GetCallerIdentityRequest, GetCallerIdentityResponse, GetCallerIdentityError,
+        GetFederationTokenRequest, GetFederationTokenResponse, GetFederationTokenError,
+        GetSessionTokenRequest, GetSessionTokenResponse, GetSessionTokenError,
+        StsClient};
+
+    pub const DEFAULT_DURATION_SECONDS: i32 = 3600;
+    pub const DEFAULT_ROLE_DURATION_SECONDS: i32 = 900;
+
+    pub trait NewAwsCredsForStsCreds {
```

Comment:
I'd love to see comments on anything publicly exported so it'll show up in our API docs.